### PR TITLE
database

### PR DESCRIPTION
**Pull request by pvhiep96**
Create table :
- User
- FriendRequest
- Friendship
- UserRoom
- Room
- Message

### CODE REVIEW

**Comment by dinhthai2206** [comment]
LGTM!

File: app/models/friendship.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Friendship < ApplicationRecord
+  belongs_to :user
+  belongs_to :friend, class_name: 'User'
+
+  validates :user_id, uniqueness: { scope: :friend_id }
```
Comment (vuhuutuan123) [inline app/models/friendship.rb]: `{ user_id: 1, friend_id: 2 }`  is existing in database.
So what if data like `{ user_id: 2, friend_id: 1 }` wanna create ? , Is it valid ?

File: app/models/permision.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Permision < ApplicationRecord
+  belongs_to :users
+  belongs_to :room
+
+  validates :user_id, uniqueness: { scope: :room_id }
```
Comment (vuhuutuan123) [inline app/models/permision.rb]: what if you are `admin` but you are muted ( only reader ) ?

File: app/models/friendship.rb (hunk 1 above)
Comment (pvhiep96) [reply app/models/friendship.rb]: when 2 users add friend, 2 records of friendship are create. So I think it enough :bowing_man:

File: app/models/permision.rb (hunk 2 above)
Comment (pvhiep96) [reply app/models/permision.rb]: I don't understand what you mean

I think admin can't be muted . All of admin has same permission (admin can't mute other admin)

**Comment by vuhuutuan123** [comment]
LGTM

**Pull request merged**